software/firmware/bootloader.py

```diff
@@ -175,3 +175,12 @@ def main(self):
                 self.show_error(
                     "Crash", f"{err.__class__.__name__[0:MAX_CHARS]}\n{str(err)[0:MAX_CHARS]}", -1
                 )
+
+                # Log the crash to a file for later analysis/recovery
+                try:
+                    with open("last_crash.log", "w") as log_file:
+                        log_file.write(f"{time.ticks_ms()}: {err}\n")
```

Comment:
[required] As written this code provides very little information in the log file, just the timestamp and error message. Since we're not constrained to the screen here, we should at least print the traceback with:

```python
sys.print_exception(err, log_file)
```

See https://docs.micropython.org/en/v1.20.0/library/sys.html#sys.print_exception

Reply:
Done